main/zoe/guide/contracts/loan.md

```diff
@@ -0,0 +1,143 @@
+of liquidation triggers with the `priceAuthority` results in a error. In
+that case, we have no choice but to give the collateral to the lender.
+The borrower has already exited with their loan.)
+
+The lender will want the interest earned from the loan + their refund
+or the results of the liquidation. If the price of collateral drops
+before we get the chance to liquidate, the total payout could be zero.
+Therefore, the lender cannot `want` anything in their proposal.
+
+The lender must be able to exit on demand until borrowing occurs. If the exit rule was
+`waived`, a borrower would be able to hold on to their invitation and
+effectively lock up the lender's Loan forever.  When the borrowing
+occurs, the collateral is moved to a special collateral seat to
+prevent the lender from being able to exit with collateral before the
+contract ends through repayment or liquidation.
+
+## The Borrower
+
+The borrower receives an invitation to join, makes an offer with Zoe
+escrowing their collateral, and receives their loan. The borrower seat
+is exited at this point so the borrower gets the payout of their loan,
+but the borrower also gets an object (`borrowFacet`) as their `offerResult` that will
+let them continue to interact with the contract.
+
+Once the loan has started, the borrower can repay the loan in its
+entirety (at which the lender receives the loan amount back plus
+interest and the contract closes), or add more collateral to stave off
+liquidation.
+
+## Contract Shutdown
+
+The contract will shutdown under 3 conditions:
+1. The loan (plus interest) is repaid.
+   * In this case the lender gets the repayment and the borrower gets
+    their collateral back.
+2. The value of the collateral drops and the collateral must be
+   liquidated.
+   * In this case the lender gets the outcome of the sale of the
+     collateral, and the borrower still keeps their loan.
+3. An error occurs when trying to use the priceAuthority.
+   * In this case the lender gets the collateral, and the borrower
+     keeps their loan.
+
+## Debt and Interest Calculation 
+
+Interest is calculated and compounded when the
+`periodAsyncIterable` pushes a new value. The interest rate per period
+is defined by the `interestRate` parameter.
+
+## Scheduling Liquidation
+
+Liquidation is scheduled using the `priceAuthority` parameter.
+Specifically, the contract gets a promise resolved when the value of the
+collateral falls below a particular trigger value defined by the `mmr`
+parameter: 
+
+```js
+  const internalLiquidationPromise = E(priceAuthority).quoteWhenLT(
+    allCollateral,
+    liquidationTriggerValue,
+  );
+  internalLiquidationPromise.then(liquidate);
+```
+
+The borrower can set up their own margin calls to be forewarned of a
+potential liquidation by
+getting the [`priceAuthority`](/zoe/guide/price-authority.md) from the terms and calling
+`E(priceAuthority).quoteWhenLT(allCollateral, x)` where x is the value
```

Comment:
This would benefit from a code sample showing how to use `.then()` to invoke the caller's handler.

Reply:
Yup!